definitions/npm/reselect_v3.x.x/flow_v0.54.x-/reselect_v3.x.x.js

```diff
@@ -0,0 +1,1327 @@
+// derived from the official typescript definitions found on the reselect repository
+// https://github.com/reactjs/reselect/blob/master/src/index.d.ts
+declare module "reselect" {
+  declare type Selector<S, R> = (state: S) => R;
+
+  declare type OutputSelector<S, R, C> = Selector<S, R> & {
+    resultFunc: C,
+    recomputations: () => number,
+    resetRecomputations: () => number
+  };
+
+  declare type ParametricSelector<S, P, R> = (
+    state: S,
+    props: P,
+    ...args: any[]
+  ) => R;
```

Comment:
I see it's derived from the typescript version, but I don't get why we need to have both Selectors and ParametricSelectors. Wouldn't it be enough to have only the ParametricSelectors ? (and then rename them Selectors :) ). This is what the old libdef does.

Reply:
Also I think the "ParametricSelector" part just doesn't work :) I added a test to prove it:

```javascript
// TEST: props as second argument
const selector = createSelector(
  (state: State, props) => "foo",
  (state) => "bar",
  (result1, result2) => result1 + result2
);
// END TEST
```

You'll get this error in flow v0.54 => 0.56:

```
   Error: 0-test_reselect_3.x.x.js:107
                         v--------------
   107: const selector = createSelector(
   108:   (state: State, props) => "foo",
   109:   (state) => "bar",
   110:   (result1, result2) => result1 + result2
   111: );
        ^ function call. Could not decide which case to select
   107: const selector = createSelector(
                         ^^^^^^^^^^^^^^ callable object type
     Case 3 may work:
              v--------------
      41:     <S, R1, R2, T>(
      42:       selector1: Selector<S, R1>,
      43:       selector2: Selector<S, R2>,
      44:       combiner: (res1: R1, res2: R2) => T
      45:     ): OutputSelector<S, T, (res1: R1, res2: R2) => T>,
              -------------------------------------------------^ polymorphic type: function type. See lib: reselect_v3.x.x.js:41
     But if it doesn't, case 4 looks promising too:
              v-----------------
      47:     <S, P, R1, R2, T>(
      48:       selector1: ParametricSelector<S, P, R1>,
      49:       selector2: ParametricSelector<S, P, R2>,
      50:       combiner: (res1: R1, res2: R2) => T
      51:     ): OutputParametricSelector<S, P, T, (res1: R1, res2: R2) => T>,
              --------------------------------------------------------------^ polymorphic type: function type. See lib: reselect_v3.x.x.js:47
     Please provide additional annotation(s) to determine whether case 3 works (or consider merging it with case 4):
     108:   (state: State, props) => "foo",
                           ^^^^^ parameter `props`
```